code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm

```diff
@@ -546,3 +546,85 @@
 		return TRUE
 	else
 		return FALSE
+
+// -------------------------
+//  Seedbin
+// -------------------------
+/obj/machinery/smartfridge/seedbin
+	name = "seed bin"
+	desc = "Organised dumping ground for the starters of life."
+	icon = 'icons/obj/hydroponics/equipment.dmi'
+	icon_state = "seedbin"
+	max_n_of_items = 400
+
+/obj/machinery/smartfridge/seedbin/accept_check(obj/item/O)
+	if(istype(O, /obj/item/seeds))
+		return TRUE
+	return FALSE
+
+// ---------------------------------------
+// Update Icons for Seed Bin
+// ---------------------------------------
+/obj/machinery/smartfridge/seedbin/update_icon_state()
+	SSvis_overlays.remove_vis_overlay(src, managed_vis_overlays)
+	if(!machine_stat)
+		SSvis_overlays.add_vis_overlay(src, icon, "smartfridge-light-mask", EMISSIVE_PLANE, dir, alpha)
+		if(visible_contents)
+			switch(contents.len)
+				if(0)
+					icon_state = "[initial(icon_state)]"
+				if(1 to 50)
+					icon_state = "[initial(icon_state)]-1"
+				if(51 to 100)
+					icon_state = "[initial(icon_state)]-2"
+				if(101 to 150)
+					icon_state = "[initial(icon_state)]-3"
+				if(151 to 200)
+					icon_state = "[initial(icon_state)]-4"
+				if(201 to INFINITY)
+					icon_state = "[initial(icon_state)]-5"
+		else
+			icon_state = "[initial(icon_state)]"
+	else
+		icon_state = "[initial(icon_state)]-off"
+
+//-------------------------
+// grownbin
+//-------------------------
+/obj/machinery/smartfridge/grownbin
+	name = "Harvest bin"
+	desc = "A large box, to contain the harvest that the Earth has blessed upon you."
+	icon = 'icons/obj/hydroponics/equipment.dmi'
+	icon_state = "grownbin"
+	max_n_of_items = 1000
+
+/obj/machinery/smartfridge/grownbin/accept_check(obj/item/O)
+	if(istype(O, /obj/item/food/grown))
+		return TRUE
+	return FALSE
```

Comment:
Two things:
1. Better variable names (as per contribution guideline)
2. You also need to check for `/obj/item/grown`, since our plants are split across two separate types.
```suggestion
/obj/machinery/smartfridge/grownbin/accept_check(obj/item/inserted_item)
	if(istype(inserted_item, /obj/item/food/grown))
		return TRUE
	if(istype(inserted_item, /obj/item/grown))
		return TRUE
	return FALSE
```

Reply:
But all other smartfriges there uses O

Reply:
Because other smartfridges are much older code that no one has improved or touched since our contribution guidelines were changed.